SDA .nc file output update

kzarada opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!--- Please select appropriate Priority, Status,and Type labels-->
<!--- If you do not have permission to select labels please state which labels you would like -->

## Description
For not removing the `nc` output files of model simulations in SDA, @para2x was renaming the outputs to their name + a timestamp. But there was a bug in the code that this PR is fixing.

<!--- The SDA function was overwriting the .nc files that were created so we only got one output instead of one for each day. Fixed the overwrite part of the function to exclude the files that were just renamed.  -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## Review Time Estimate
<!---When do you want your code reviewed by?-->
- [ ] Immediately
- [ ] Within one week
- [ X] When possible
## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue) <!-- please add issue number -->
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

<!--this template is from https://www.talater.com/open-source-templates/#/page/99-->

para2x commented:
This looks good to me and it's what we discussed. Did you also get to test it ?

kzarada commented:
@para2x yes, I ran it and it worked properly

infotroph commented on modules/assim.sequential/R/sda.enkf_refactored.R:
```diff
@@ -350,19 +350,24 @@ sda.enkf <- function(settings,
         file.rename(.x ,
                     file.path(dirname(.x),
-                              paste0(gsub(" ", "", as.character(names(obs.mean)[t])),
+                              paste0("SDA_",gsub(" ", "", as.character(names(obs.mean)[t])),
```
Are all files in `files` guaranteed to live in different directories? If any of them share a `dirname`, they'll be given the same name and therefore overwrite each other. Or am I missing something?

Also, less importantly: `file.rename` accepts vectors, and `names()` is documented to return characters if it returns anything, so I *think* you can drop both `walk` and `as.character` and simplify this to

```
file.rename(
  files,
  file.path(
    dirname(files),
    paste0("SDA_", gsub(" ", "", names(obs.mean)[t]), ".nc"))
```

para2x replied:
I haven't seen any of our models to produce more than one `nc` output file per dir, but that's a good point and we can slightly modify that to use the `basename` but then we need the `walk` again something like this:
`paste0("SDA_", basename(.x), "_", names(obs.mean)[t], ".nc")`
>

para2x replied:
oh seems like the `basename` function is also vectorized and yeah seems like we don't the `walk` at all.

para2x replied:
@kzarada could you please give a try to this replacing the whole `walk` see if it works as expected :
`
file.rename(
    files,
    file.path(
       dirname(files),
       paste0("SDA_", basename(files), "_", gsub(" ", "", names(obs.mean)[t]), ".nc")
     )
 )
`

kzarada replied:
That code worked- the output files look like this 'SDA_2019.nc_2019-11-0500:00:00.nc'

kzarada replied:
I pushed the update

kzarada commented:
@infotroph those were accidentally added (I forgot to create a new branch with this update since it was still WCr SDA related)- not relevant to this bug fix but they are part of the WCr SDA. I just added a shell file to run the forecast automatically, updated the workflow to work with the automation, and increased the ensemble size

mdietze commented:
@kzarada those accidentally added files all sound like things we want to pull in, so lets go ahead and keep them in this PR.
